components/sendgrid/sendgrid.app.js

```diff
@@ -118,5 +113,429 @@ module.exports = {
+     * array with details of each contact list returned: `_metadata` with a `self` object containing
+     * a link to the contat list, a `contact_count` with the count of contacts in the list, an `id`
+     * as a unique identifier to the contat list, and the `name` of the list.
+     */
+    async getAllContactLists(maxItems) {
+      const pageSize = Math.min(maxItems, 1000);
+      const contactLists = [];
+      let url = `/v3/marketing/lists?page_size=${pageSize}`;
+      do {
+        const data  = (await this._withRetries(() => this.api().request({
+          url,
+        })))[1];
+        contactLists.push(...data.result);
+        if (!data._metadata.next)
+        {
+          break;
+        }
+        url = data._metadata.next.replace("https://api.sendgrid.com", "");
+      } while (true && contactLists.length < maxItems);
+      return contactLists.slice(0, maxItems);
+    },
+    /**
+     * Lists all email addresses that are currently the associated account blocks list.
+     *
+     * @param {integer}  startTime start of the time range in unix timestamp when a block was
+     * created (inclusive).
+     * @param {integer}  endTime end of the time range in unix timestamp when a block was
+     * created (inclusive).
+     * @param {integer}  numberOfBlocks the number of blocks to return.
+     * @returns {{created: number, email: string, reason: string, status: string}: array} an array
+     * with details of each block returned: `created` for unix timestamp for when the block record
+     * was created at SendGrid, `email` for the email address that was added to the block  list,
+     * `reason` with the reason for the block, and the `status` of the block.
+     */
+    async listBlocks(startTime, endTime, maxItems) {
+      const pageSize = Math.min(maxItems, 100);
+      const blocks = [];
+      let url = `/v3/suppression/blocks?limit=${pageSize}`;
+      let config = {
+        method: "GET",
+        url,
+        start_time: startTime,
+        end_time: endTime,
+      };
+      let lastIteration = false;
+      do {
+        const data  = (await this._withRetries(() => this.api().request(config)));
+        blocks.push(...data[1]);
+        if (lastIteration) {
+          break;
+        }
+        const links = data[0].headers.link.split(",");
+        url = "";
+        links.forEach( (link) => {
+          let next = "";
+          let last = "";
+          if ( link.indexOf("next") > -1) {
+            const idx = link.indexOf(";");
+            next = link.substring(0, idx).replace("<", "")
+              .replace(">", "");
+            url = next;
+          }
+          if ( link.indexOf("last") > -1) {
+            const idx = link.indexOf(";");
+            last = link.substring(0, idx).replace("<", "")
+              .replace(">", "");
+          }
+          const nextArr = [];
+          nextArr.push(next);
+          if (nextArr.includes(last)) {
+            lastIteration = true;
+          }
```

Comment:
it seems like you just need something like
```suggestion
          lastIteration = next === last;
```